syft/pkg/license.go

```diff
@@ -0,0 +1,34 @@
+package pkg
+
+import (
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/license"
+	"github.com/anchore/syft/syft/source"
+)
+
+type License struct {
+	Value          string          `json:"value"`
+	SPDXExpression string          `json:"spdx-expression"`
+	Type           license.Type    `json:"type"`
+	URL            string          `json:"url"`      // external sources
+	Location       source.Location `json:"location"` // on disk declaration
+}
+
+func NewLicense(value string, url string, location source.Location) License {
```

Comment:
it seems that url is blank in many cases, but this also looks like a future extension point. This hints at providing tailor-made constructors:
```
func NewLicenseFromLocation(value string, location source.Location) License {}

func NewLicenseFromURL(value, url string) License {}
```